### DISCUSSION TITLE
FR: Fullscreen Matrix

### DISCUSSION THREAD

**Discussion by richhayler** [craftcms/cms#8695] [Ideas]
As a content author I love being able to strip away any of the unnecessary UI so I can focus completely on the content i’m editing.

When editing matrix blocks for an entry It would be awesome if the block had a fullscreen toggle and then arrow keys to move to the next or previous blocks while still in fullscreen mode.

This feature would:

- Provide a new way for authors to edit matrix content that strips away unnecessary UI
- Reduces the need to scroll
- Provide a larger canvas for editing content (great for smaller screen sizes, e.g. iPad)

**Comment by moacode** [comment]
@richhayler I've created a first cut of a plugin to help with this. You can test the alpha here: https://github.com/thejoshsmith/craft-expanded-matrix (only supports keyboard left/right arrows atm).

Still to do is:
- Support swiping on mobile devices.
- Refine styles + add left/right clickable arrows.
- Mobile testing.

I'm thinking of creating a premium version that can additionally display the blocks in a configurable grid. The idea being the user can specify a width for each block (i.e. 25%) and add blocks or reorder them from this view.

Let me know if you have any thoughts/feedback!

**Comment by moacode** [comment]
Now on the plugin store (free): https://plugins.craftcms.com/expanded-matrix

**Comment by richhayler** [comment]
Fantastic work @thejoshsmith I installed this straight away and I'm loving it. It works seamlessly for me and makes a massive difference being able to edit larger matrix content blocks. Thank you for this!
